src/ashley/machina_extensions/forum_conversation/views.py

```diff
@@ -0,0 +1,47 @@
+"""
+    Forum conversation views
+    ========================
+    This module defines views provided by the ``forum_conversation`` application.
+"""
+
+from django.http import HttpResponseRedirect
+from django.shortcuts import get_object_or_404
+from machina.apps.forum_conversation.views import TopicCreateView as BaseTopicCreateView
+from machina.core.db.models import get_model
+
+from .signals import topic_created
+
+Topic = get_model("forum_conversation", "Topic")
+
+
+class TopicCreateView(BaseTopicCreateView):
+    """Creates a topic within a forum."""
+
+    view_signal = topic_created
+
+    def form_valid(self, post_form, attachment_formset, poll_option_formset, **kwargs):
+        """Raise topic created signal on successful topic creation"""
+
+        response = super().form_valid(
+            post_form, attachment_formset, poll_option_formset, **kwargs
+        )
+
+        if self.forum_post.topic.pk is not None:
+            topic = get_object_or_404(
+                Topic,
+                pk=self.forum_post.topic.pk,
+            )
+            self.send_signal(self.request, response, topic)
```

Comment:
Since the topic has been created, there is no need to fetch it once again?

```suggestion
            self.send_signal(self.request, response, self.forum_post.topic)
```